src/Zygote.jl

```diff
@@ -43,15 +43,15 @@ include("profiler/Profile.jl")
   include("flux.jl")
 end
 
-precompile() = include(joinpath(@__DIR__, "precompile.jl"))
+_precompile() = include(joinpath(@__DIR__, "precompile.jl"))
 
-# precompile()
-@init Requires.isprecompiling() || precompile()
+# _precompile()
+#@init Requires.isprecompiling() || _precompile()
 
 # helps to work around 265-y issues
 function refresh()
   include(joinpath(@__DIR__, "compiler/interface2.jl"))
-  precompile()
+  _precompile()
   return
```

Comment:
What do these lines do here? Are they precompiling?

Reply:
That's an ongoing discussion in #607 Seems like it's an antipattern that should be removed.
I commented it out to test it in this PR. However I don't know when the `refresh()` function is called. I suspect it can be removed, but thought that was better to do in #607 

However, I'm now using the contents of `joinpath(@__DIR__, "precompile.jl")` as the snoop code

Reply:
Let's move any precompiling preparation code to the snooping code instead. They shouldn't be in the main source file.